packages/siop-oid4vp/lib/authorization-response/OpenID4VP.ts

```diff
@@ -278,21 +283,25 @@ export const assertValidVerifiablePresentations = async (args: {
     hasher?: Hasher
   }
 }) => {
+  const presentationsArray = Array.isArray(args.presentations) ? args.presentations : [args.presentations]
+  const nrOfMdocsPresentations = presentationsArray.filter(presentation => presentation.format === 'mso_mdoc').length
+  const presentationsWithFormat = presentationsArray.filter(presentation => presentation.format !== 'mso_mdoc')
   if (
     (!args.presentationDefinitions || args.presentationDefinitions.filter((a) => a.definition).length === 0) &&
-    (!args.presentations || (Array.isArray(args.presentations) && args.presentations.filter((vp) => vp.presentation).length === 0))
+    (!presentationsWithFormat || (Array.isArray(presentationsWithFormat) && presentationsWithFormat.filter((vp) => vp.presentation).length === 0))
   ) {
     return
   }
   PresentationExchange.assertValidPresentationDefinitionWithLocations(args.presentationDefinitions)
-  const presentationsWithFormat = args.presentations
 
   if (
     args.presentationDefinitions &&
     args.presentationDefinitions.length &&
     (!presentationsWithFormat || (Array.isArray(presentationsWithFormat) && presentationsWithFormat.length === 0))
   ) {
-    throw new Error(SIOPErrors.AUTH_REQUEST_EXPECTS_VP)
+    if(nrOfMdocsPresentations === 0) {
```

Comment:
Whut? At least make a comment on why something like this would be needed

Reply:
I guess I just wanted to make assertValidVerifiablePresentations not crash when running mdocs in which case presentationsWithFormat will be empty. (But the presentationDefinitions are still asserted)
Will add comment

Reply:
Well I believe current approach is wrong in any case. This triggers if there is no presentation. But then we only provide an error if there are no mdoc presentations. That doesn't make sense to me. 

That is always true anyway probably given mdoc is being filtered out above.